core/generator/generator.go

```diff
@@ -75,7 +75,16 @@ func Generate(
 		log.Fatal(ctx, log.KeyError, err)
 	}
 	if b != nil && (g.latestBlock == nil || b.Height == g.latestBlock.Height+1) {
-		s := state.Copy(g.latestSnapshot)
+		var s *state.Snapshot
+		if g.latestSnapshot == nil {
```

Comment:
What about passing it explicitly as an argument? We have a blockchain ID in cored when we launchConfiguredCore. That will call `leader.Run`, which in turn calls either `fetch.Fetch` or `generator.Generate`. Those are the two functions that call `Recover`.